typing error with the latest version 4.1.0

TypeScript error occurred after upgrading with the latest version 4.1.0.

Here is the modified version of sample code in readme.md

```ts
const pThrottle = require('p-throttle');

const now = Date.now();

const throttle = pThrottle({
  limit: 2,
  interval: 1000,
});

const func = (index: number, tag: string): Promise<string> => {
  const secDiff = ((Date.now() - now) / 1000).toFixed();
  return Promise.resolve(`${tag} ${index}: ${secDiff}s`);
};

const throttled = throttle<number, string>(func);
```

I got a following error.

```
error TS2345: Argument of type '(index: number, tag: string) => Promise<string>' is not assignable to parameter of type '(...arguments: number[]) => string'.
  Types of parameters 'tag' and 'arguments' are incompatible.
    Type 'number' is not assignable to type 'string'.
```

It seems that current definition assumes all types of arguments are [the same](https://github.com/sindresorhus/p-throttle/pull/23/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R13-R15).

```ts
const throttled = throttle(func);
```

```
error TS2345: Argument of type '(index: number, tag: string) => Promise<string>' is not assignable to parameter of type '(...arguments: (string | number)[]) => Promise<string>'.
  Types of parameters 'index' and 'arguments' are incompatible.
    Type 'string | number' is not assignable to type 'number'.
      Type 'string' is not assignable to type 'number'.
```

> The first type argument of throttle should be a tuple of the arguments.

So [this](https://github.com/sindresorhus/p-throttle/pull/23/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R13-R15) should be
```
	type ThrottledFunction<Argument, ReturnValue> = ((
		...arguments: Argument // <--
	) => PromiseResolve<ReturnValue>) & {```

I tried

```ts
const throttled = throttle(func);
```

but got

```
error TS2345: Argument of type '(index: number, tag: string) => Promise<string>' is not assignable to parameter of type '(...arguments: (string | number)[]) => Promise<string>'.
  Types of parameters 'index' and 'arguments' are incompatible.
    Type 'string | number' is not assignable to type 'number'.
      Type 'string' is not assignable to type 'number'.
```

> The first type argument of throttle should be a tuple of the arguments.

I see. I just created a PR #29 . Would you mind to review?

--- a/index.d.ts
+++ b/index.d.ts
@@ -10,8 +10,8 @@ declare class AbortErrorClass extends Error {
 type PromiseResolve<ValueType> = ValueType extends PromiseLike<infer ValueType> ? Promise<ValueType> : Promise<ValueType>;
 
 declare namespace pThrottle {
-	type ThrottledFunction<Argument, ReturnValue> = ((
-		...arguments: Argument[]
+	type ThrottledFunction<Argument extends readonly unknown[], ReturnValue> = ((
+		...arguments: Argument
 	) => PromiseResolve<ReturnValue>) & {
 		/**
 		Whether future function calls should be throttled or count towards throttling thresholds.
@@ -78,7 +78,7 @@ declare const pThrottle: {
 	*/
 	(
 		options: pThrottle.Options
-	): <Argument, ReturnValue>(function_: (...arguments: Argument[]) => ReturnValue) => pThrottle.ThrottledFunction<Argument, ReturnValue>;
+	): <Argument extends readonly unknown[], ReturnValue>(function_: (...arguments: Argument) => ReturnValue) => pThrottle.ThrottledFunction<Argument, ReturnValue>;
 };
 
 export = pThrottle;

--- a/index.test-d.ts
+++ b/index.test-d.ts
@@ -26,10 +26,10 @@ const strictThrottledLazyUnicorn = pThrottle({
 
 expectType<AbortError>(new AbortError());
 
-expectType<ThrottledFunction<string, string>>(throttledUnicorn);
-expectType<ThrottledFunction<string, Promise<string>>>(throttledLazyUnicorn);
-expectType<ThrottledFunction<string, string>>(strictThrottledUnicorn);
-expectType<ThrottledFunction<string, Promise<string>>>(strictThrottledLazyUnicorn);
+expectType<ThrottledFunction<[string], string>>(throttledUnicorn);
+expectType<ThrottledFunction<[string], Promise<string>>>(throttledLazyUnicorn);
+expectType<ThrottledFunction<[string], string>>(strictThrottledUnicorn);
+expectType<ThrottledFunction<[string], Promise<string>>>(strictThrottledLazyUnicorn);
 
 throttledUnicorn.abort();
 throttledLazyUnicorn.abort();

--- a/index.test-d.ts
+++ b/index.test-d.ts
@@ -24,6 +24,11 @@ const strictThrottledLazyUnicorn = pThrottle({
 	strict: true
 })(async (index: string) => '🦄');
 
+const throttledTaggedUnicorn = pThrottle({
+	limit: 1,
+	interval: 1000
+})((index: number, tag: string) => `${tag}: 🦄`);
+
 expectType<AbortError>(new AbortError());
 
 expectType<ThrottledFunction<[string], string>>(throttledUnicorn);

--- a/index.test-d.ts
+++ b/index.test-d.ts
@@ -35,10 +35,12 @@ expectType<ThrottledFunction<[string], string>>(throttledUnicorn);
 expectType<ThrottledFunction<[string], Promise<string>>>(throttledLazyUnicorn);
 expectType<ThrottledFunction<[string], string>>(strictThrottledUnicorn);
 expectType<ThrottledFunction<[string], Promise<string>>>(strictThrottledLazyUnicorn);
+expectType<ThrottledFunction<[number, string], Promise<string>>>(throttledTaggedUnicorn);
 
 throttledUnicorn.abort();
 throttledLazyUnicorn.abort();
 strictThrottledUnicorn.abort();
 strictThrottledLazyUnicorn.abort();
+throttledTaggedUnicorn.abort();
 
 expectType<boolean>(throttledUnicorn.isEnabled);

